opensquirrel/exporter/quantify_scheduler_exporter.py

```diff
@@ -86,21 +82,27 @@ def visit_controlled_gate(self, g: ControlledGate) -> None:
             )
             return
 
-        raise _unsupported_gates_exception
+        raise UnsupportedGateError(g)
 
 
 def export(circuit: Circuit) -> "quantify_scheduler.Schedule":
     if "quantify_scheduler" not in globals():
 
         class QuantifySchedulerNotInstalled:
             def __getattr__(self, attr_name: Any) -> None:
-                raise ImportError("quantify-scheduler is not installed, or cannot be installed on your system")
+                raise ModuleNotFoundError("quantify-scheduler is not installed, or cannot be installed on your system")
 
         global quantify_scheduler
         quantify_scheduler = QuantifySchedulerNotInstalled()
         global quantify_scheduler_gates
         quantify_scheduler_gates = QuantifySchedulerNotInstalled()
 
     schedule_creator = _ScheduleCreator(circuit.qubit_register_name)
-    circuit.ir.accept(schedule_creator)
+    try:
+        circuit.ir.accept(schedule_creator)
+    except UnsupportedGateError as e:
+        raise CircuitExportError(
+            f"cannot export circuit: {e} Decompose all gates to the Quantify-scheduler gate set first (rxy, rz, cnot, "
+            "cz)"
```

Comment:
```suggestion
            f"cannot export circuit: {e}. "
            "Decompose all gates to the Quantify-scheduler gate set first (rxy, rz, cnot, cz)"
```

Split into two self-contained lines just for reader's clarity.
And added a dot after the error message.